react-app/src/components/BaseModal/index.js

```diff
@@ -0,0 +1,39 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+import styles from './index.module.scss';
+
+export default class BaseModal extends PureComponent {
+  static propTypes = {
+    /**
+     * Whenever the modal is visible or not
+     */
+    visible: PropTypes.bool.isRequired,
+
+    /**
+     * The element to use as the modal content
+     */
+    content: PropTypes.element.isRequired
+  };
+
+  _getModal = () => {
+    return (
+      <div>
+        <div className={styles.opacityLayer} />
+        <div className={styles.modal}>
+          {this.props.content}
+        </div>
+      </div>
+    )
+  }
+
+  render() {
+    const modal = (this.props.visible) ? this._getModal() : null;
+
+    return (
+      <div>
```

Comment:
I think that this component will be much simpler if you write it this way:

```jsx
  render() {
    if (!this.props.visible) return null;

    return (
      <div>
        <div className={styles.opacityLayer} />
        <div className={styles.modal}>
          {this.props.content}
        </div>
      </div>
    );
  }
```

Note that I use a simple condition and return soon instead of a multiple option condition. This allows me to avoid some nesting if we were to use an `if`/`else` construction.
Extracting of part of the component work into a function is a good idea. On this case, though, the `_getModal` function does all the work that the component needs, so I've moved its code into the `render` function.

Reply:
Done